server/src/main/java/com/cloud/hypervisor/HypervisorGuruBase.java

```diff
@@ -215,7 +215,7 @@ protected VirtualMachineTO toVirtualMachineTO(VirtualMachineProfile vmProfile) {
 
         to.setNics(nics);
         to.setDisks(vmProfile.getDisks().toArray(new DiskTO[vmProfile.getDisks().size()]));
-
+        // FR37 if this is a new VM to be deployed as is from a template we need to pass the mary storage somehow, now only for actual disks is a primary storage passed
```

Comment:
not sure but i think we can 
```suggestion
```
if not fully handled yet we might want to change to 
```suggestion
        // If this is a new VM to be deployed as is from a template we need to pass the primary storage somehow, now only for actual disks is a primary storage passed
```